### ISSUE TITLE
Recursive Schema - Cannot locate inline $id references

### ISSUE THREAD

**Issue by sinclairzx81** [fastify/fast-json-stringify#410]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

fast-json-stringify@3.1.0

### Plugin version

n/a

### Node.js version

16.13.1

### Operating system

Windows

### Operating system version (i.e. 20.04, 11.3, 10)

10

### Description

Hi. I'm currently looking at recursive schema composition in TypeBox, and have been carrying out a few tests in Fastify. Have run into an issue in `fast-json-stringify` where it seems unable to locate a previously defined identifier for an inline (non-external) reference.

Also may have picked up on a small type error where `$id` is not a valid property which makes passing top level `$id` for general schemas not possible.

![image](https://user-images.githubusercontent.com/3048342/165516393-b4501388-e575-4bd3-b932-369a5caefa8f.png)

Related Issue

https://github.com/fastify/fast-json-stringify/issues/384

### Steps to Reproduce

The following is the minimal repro replicating the following schema type.
```typescript
interface Node {
  id: string
  nodes: Node[]
}
```
```typescript
import fastJson from 'fast-json-stringify'

const stringify = fastJson({
    "$id": "Node",
    "type": "object",
    "properties": {
        "id": {
            "type": "string"
        },
        "nodes": {
            "type": "array",
            "items": {
                "$ref": "Node"
            }
        }
    },
    "required": [
        "id",
        "nodes"
    ]
})
```
Results in the following error
```
        schema = externalSchema[ref[0]];
                               ^

TypeError: Cannot read properties of undefined (reading 'Node')
```

### Expected Behavior

fast-json-stringify should be able to accept schemas of this form.

**Comment by sinclairzx81** [comment]
@mcollina Hi, this issue should be resolved by https://github.com/fastify/fast-json-stringify/pull/411. I may be able to take a look at the related issue https://github.com/fastify/fast-json-stringify/issues/384 in a subsequent PR in a few months time.

**Comment by climba03003** [comment]
You can link to the issue when you use the below keyword in PR description. It will automatically close the linked issue when the PR merged.
- close
- closes
- closed
- fix
- fixes
- fixed
- resolve
- resolves
- resolved

For example: `Resolved #410`

**Comment by sinclairzx81** [comment]
@climba03003 Thanks :)
